### DISCUSSION TITLE
How to enable protobuf?

### DISCUSSION THREAD

**Discussion by Helmet93** [netdata/netdata#12275] [Q&A] [answered]
Hello,
I received the notice that after the 15th of March my node will not be monitored by Netdata cloud. I ran an update and am on the latest version, but however when I run `netdata -W buildinfo` I still see I have:
`Libraries:
protobuf: NO`

How can I enable protobuf?

I originally used the Kickstart script to install and used that to update as well.

I am on Netdata version `v1.33.1-78-nightly`

**Comment by MrZammler** [comment]
Hi @Mossy93 ! Thanks! Can you please post the full output of `netdata -W buildinfo` ?

**Reply by Helmet93** [reply]
```cmd
Version: netdata v1.33.1-86-nightly
Configure options:  '--prefix=/usr' '--sysconfdir=/etc' '--localstatedir=/var' '--libexecdir=/usr/libexec' '--libdir=/usr/lib' '--with-zlib' '--with-math' '--with-user=netdata' '--disable-ml' 'CFLAGS=-O2' 'LDFLAGS='
Install type: kickstart-build
Features:
    dbengine:                   YES
    Native HTTPS:               YES
    Netdata Cloud:              YES 
    ACLK Next Generation:       YES
    ACLK-NG New Cloud Protocol: NO
    ACLK Legacy:                NO
    TLS Host Verification:      YES
    Machine Learning:           NO
    Stream Compression:         NO
Libraries:
    protobuf:                NO
    jemalloc:                NO
    JSON-C:                  YES
    libcap:                  NO
    libcrypto:               YES
    libm:                    YES
    tcalloc:                 NO
    zlib:                    YES
Plugins:
    apps:                    YES
    cgroup Network Tracking: YES
    CUPS:                    NO
    EBPF:                    YES
    IPMI:                    NO
    NFACCT:                  NO
    perf:                    YES
    slabinfo:                YES
    Xen:                     NO
    Xen VBD Error Tracking:  NO
Exporters:
    AWS Kinesis:             NO
    GCP PubSub:              NO
    MongoDB:                 NO
    Prometheus Remote Write: NO
```

**Comment by MrZammler** [comment, marked as answer]
Thanks! Do you have a C++ compiler on your machine? Can you post your distribution you're using?

**Reply by Helmet93** [reply]
I am using DietPi 8.1.2

Not sure about the C++ compiler. I'd have to look into that. I've never explicitly installed one though

**Reply by MrZammler** [reply]
Most likely installing a C++ compiler will allow to build protobuf (I suspect that's the reason it's missing). However, the load of the compile could be substantial for a raspberry pi (which model btw?). If that's too much for the raspberry pi to handle, you could also check for using a [native-package](https://learn.netdata.cloud/docs/agent/packaging/installer/methods/kickstart#native-packages).

**Reply by MrZammler** [reply]
In case of the DietPi (which is based on Debian), using `sudo apt-get install build-essential` should install what is needed.

**Reply by Helmet93** [reply]
I installed `build-essential` and rebooted. After that I ran the install script, but it still shows protobuf as off.

**Reply by Helmet93** [reply]
It's a Pi 3b+ 1GB, by the way

**Reply by MrZammler** [reply]
When you say you run the install script, you mean the kickstart script? Can you please check again for protobuf support? (It might have compiled it during a nightly update). Can you please re-run it and post the output?

If not, then the only choice is to use a binary build (or install protobuf system wide to use that instead).

**Reply by Helmet93** [reply]
Yes, the Kickstart script.

I just checked `netdata -W buildinfo` and got
`Version: netdata v1.33.1-99-nightly
Configure options:  '--prefix=/usr' '--sysconfdir=/etc' '--localstatedir=/var' '--libexecdir=/usr/libexec' '--libdir=/usr/lib' '--with-zlib' '--with-math' '--with-user=netdata' '--disable-ml' '--with-bundled-protobuf' 'CFLAGS=-O2' 'LDFLAGS='
Install type: kickstart-build
Features:
    dbengine:                   YES
    Native HTTPS:               YES
    Netdata Cloud:              YES
    ACLK Next Generation:       YES
    ACLK-NG New Cloud Protocol: YES
    ACLK Legacy:                NO
    TLS Host Verification:      YES
    Machine Learning:           NO
    Stream Compression:         NO
Libraries:
    protobuf:                YES (bundled)
    jemalloc:                NO
    JSON-C:                  YES
    libcap:                  NO
    libcrypto:               YES
    libm:                    YES
    tcalloc:                 NO
    zlib:                    YES
Plugins:
    apps:                    YES
    cgroup Network Tracking: YES
    CUPS:                    NO
    EBPF:                    YES
    IPMI:                    NO
    NFACCT:                  NO
    perf:                    YES
    slabinfo:                YES
    Xen:                     NO
    Xen VBD Error Tracking:  NO
Exporters:
    AWS Kinesis:             NO
    GCP PubSub:              NO
    MongoDB:                 NO
    Prometheus Remote Write: NO`

**Reply by Helmet93** [reply]
Netdata Cloud still claims I don't have protobuf support on my node though

**Reply by Ferroin** [reply]
Unless something has changed very recently, it takes some time for the Cloud to recognize properly that a node has protobuf support. The Cloud team has been working on improving this, but for the moment just from the buildinfo output it _looks_ like you should be fine.

**Reply by Helmet93** [reply]
Cloud is good now. I'm getting readings from my node and the warning is gone. Thank you.

**Comment by Helmet93** [comment]
Is there something in `netdata.conf` I might need to change or enable?

**Reply by MrZammler** [reply]
Hi @Mossy93 , no, there's not something in netdata.conf to edit.
